src/libsyntax/ext/tt/macro_parser.rs

```diff
@@ -696,10 +696,17 @@ pub fn parse(
             } else {
                 return Failure(parser.span, token::Eof);
             }
+        } else {
```

Comment:
Just curious, why put this in an `else`?

Reply:
**Short answer:** To satisfy `dropck`.

---

**Long answer:**

I primarily saw two options here:

**Drop `eof_items` at the beginning of the one branch that is currently affected by the `drop`.** -- This had the disadvantage that future refactorings could easily introduce similar performance regressions (possibly under different conditions).

**Drop `eof_items` as soon as possible.** -- This required splitting the large  `if {} else if {} else if {} ...` into two.  I saw this as the lesser of two evils, despite the following disadvantages:
* `eof_items` must be awkwardly dropped inside an `else`, since it was moved in the `if` branch.
* It takes a bit of careful reading to realize that the `if` branch unconditionally diverges, especially now that it has an `else` branch. For this reason, I added the assertion two lines below.

Reply:
Oops.  Right after posting that I decided to double-check on the playground, and it turns out that move analysis is smart enough that the `else` is not necessary.